spec/plot_spec.rb

```diff
@@ -41,49 +41,59 @@
-      let(:df) do
-        Daru::DataFrame.new(
-          {
-            a: [1, 2, 3, 4, 5, 6],
-            b: [1, 5, 2, 5, 1, 0],
-            c: [1, 6, 7, 2, 6, 0]
-          }, index: 'a'..'f'
-        )
-      end
-      let(:dv) { Daru::Vector.new [1, 2, 3] }
-      let(:plot_df) { Daru::View::Plot.new(df, type: :line, x: :a, y: :c) }
-      let(:plot_dv) { Daru::View::Plot.new(dv, type: :line) }
-      let(:plot_array) { Daru::View::Plot.new([1, 2, 3], type: :line) }
-      it 'check plotting library' do
-        expect(lib).to eq(:highcharts)
-      end
+  before { Daru::View.plotting_library = :highcharts }
+  let(:lib) { Daru::View.plotting_library }
+  let(:df) do
+    Daru::DataFrame.new(
+      {
+        a: [1, 2, 3, 4, 5, 6],
+        b: [1, 5, 2, 5, 1, 0],
+        c: [1, 6, 7, 2, 6, 0]
+      }, index: 'a'..'f'
+    )
+  end
+  let(:dv) { Daru::Vector.new [1, 2, 3] }
+  let(:plot_df) { Daru::View::Plot.new(df, type: :line, x: :a, y: :c) }
+  let(:plot_dv) { Daru::View::Plot.new(dv, type: :line) }
+  let(:plot_array) { Daru::View::Plot.new([1, 2, 3], type: :line) }
+  context 'initialize with HighCharts library'  do
+    it 'check plotting library' do
+      expect(lib).to eq(:highcharts)
+    end
 
-      it 'Highcharts chart using Array' do
-        expect(plot_array).to be_a Daru::View::Plot
-        expect(plot_array.chart).to be_a LazyHighCharts::HighChart
-        expect(plot_array.data).to eq [1, 2, 3]
-        expect(plot_array.options).to eq type: :line
-      end
+    it 'Highcharts chart using Array' do
+      expect(plot_array).to be_a Daru::View::Plot
+      expect(plot_array.chart).to be_a LazyHighCharts::HighChart
+      expect(plot_array.data).to eq [1, 2, 3]
+      expect(plot_array.options).to eq type: :line
+    end
 
-      it 'Highcharts chart using DataFrame' do
-        expect(plot_df).to be_a Daru::View::Plot
-        expect(plot_df.chart).to be_a LazyHighCharts::HighChart
-        expect(plot_df.data).to eq df
-        expect(plot_df.options).to eq type: :line, x: :a, y: :c
-      end
+    it 'Highcharts chart using DataFrame' do
+      expect(plot_df).to be_a Daru::View::Plot
+      expect(plot_df.chart).to be_a LazyHighCharts::HighChart
+      expect(plot_df.data).to eq df
+      expect(plot_df.options).to eq type: :line, x: :a, y: :c
+    end
 
-      it 'Highcharts chart using Vector' do
-        expect(plot_dv).to be_a Daru::View::Plot
-        expect(plot_dv.chart).to be_a LazyHighCharts::HighChart
-        expect(plot_dv.data).to eq dv
-        expect(plot_dv.options).to eq type: :line
-      end
+    it 'Highcharts chart using Vector' do
+      expect(plot_dv).to be_a Daru::View::Plot
+      expect(plot_dv.chart).to be_a LazyHighCharts::HighChart
+      expect(plot_dv.data).to eq dv
+      expect(plot_dv.options).to eq type: :line
     end
   end # initialize context end
+
+  context '#export_type' do
+    it "should generate the valid script to export the chart" do
```

Comment:
It would be better, if you write specs for each export type and check for saved file(png, pdf, ..)

Reply:
This method generates the JS code in String format (or loads it in IRuby) to download the chart. I have updated the documentation of it. It will export the chart when it will run in web frameworks or IRuby notebook. I will write the specs for each export type. Should I write them in the same way or is there any other way possible?